src/fides/api/ops/api/v1/endpoints/policy_endpoints.py

```diff
@@ -148,6 +148,75 @@ def create_or_update_policies(
     )
 
 
+def get_rule_or_error(
+    db: Session, policy_key: FidesOpsKey, rule_key: FidesOpsKey
+) -> Rule:
+    """
+    Helper method to load Rule or throw a 404 if it can't be found.
+    Also throws a 404 if a `Policy` with the given key can't be found.
+    """
+    logger.info("Finding rule with key '{}'", rule_key)
+    policy = get_policy_or_error(db, policy_key)
+    rule = Rule.filter(
+        db=db,
+        conditions=((Rule.policy_id == policy.id) & (Rule.key == rule_key)),
+    ).first()
+    if not rule:
+        raise HTTPException(
+            status_code=HTTP_404_NOT_FOUND,
+            detail=f"No Rule found for key {rule_key}.",
```

Comment:
good call out, thanks! while we're on it, do you think this is the preferred implementation that i have here? i.e. since the rule key is technically a unique column, we could just do a rule lookup only on its `key` and not look at the `policy` in this endpoint...but i decided on making it `policy`-specific, since we can only define rules via the `policy` they're attached to, and that relationship is one-to-many (one `policy` to many `rule`s, but a `rule` can only be associated with one `policy`)

Reply:
I like what you have here, nesting it under the policy, for reasons you described